src/main/scala/impl/group5/NumberOfLoopsAnalysis.scala

```diff
@@ -0,0 +1,128 @@
+package org.tud.sse.metrics
+package impl.group5
+
+import analysis.{ClassFileAnalysis, MetricValue}
+import input.CliParser.OptionMap
+
+import org.opalj.br.ClassFile
+import org.opalj.br.analyses.Project
+
+import java.net.URL
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+/**
+ * Die Klasse NumberOfLoopsAnalysis stellt die Implementierung  der Metrik Number of Loops (Loop) da.
+ * Diese Metrik zählt für jede Methode die Anzahl der Schleifen und gibt sie aus.
+ *
+ * Die Default Einstellung ist, dass nur die Methoden ausgeben werden, in denen sich Loops befinden. Die anderen Methoden in denen Loops 0 ist werden weggelassen
+ *
+ * Optional CLI argument:
+ *  - --only-class Es wird nur die Anzahl an Loops in einer Klasse angezeigt und nicht für jede Methode
+ *  - --out-all-loops Es werden alle Methoden ausgeben, auch die mit Loop 0.
+ *  - --no-class Es wird nicht die gesamte Zahl der Loops einer Klasse ausgeben
+ *  - --only-class-with-loops Gibt nur Klassen mit Loops aus.
+ */
+
+class NumberOfLoopsAnalysis extends ClassFileAnalysis {
+
+  private val onlyClassSymbol: Symbol = Symbol("only-class")
+  private val outAllSymbol: Symbol = Symbol("out-all-loops")
+  private val noClassSymbol: Symbol = Symbol("no-class")
+  private  val onlyClassWithLoopsSymbol: Symbol = Symbol("only-class-with-loops")
+
+  /**
+   * Die Methode führt die Metrik Loop aus.
+   * @param classFile Ist ein Interface von Projekt
+   * @param project Bekommt die vom Framework gelesende jar file in einer Präsentationsform
+   * @param customOptions Einstellungs Möglichkeiten der Analyse
+   * @return Das Ergebniss wird in der Liste für die Ausgabe gespeichert
+   */
+  override def analyzeClassFile(classFile: ClassFile, project: Project[URL], customOptions: OptionMap): Try[Iterable[MetricValue]] = Try {
+
+    val outAllMethode = customOptions.contains(outAllSymbol)
+    val onlyClass = customOptions.contains(onlyClassSymbol)
+    val noClass = customOptions.contains(noClassSymbol)
+    val onClassLoops = customOptions.contains(onlyClassWithLoopsSymbol)
+
+    if(noClass &&(onlyClass||onClassLoops))
+      {
+        log.error("Fehler! Argumente sind inkompatibel")
+      }
+
```

Comment:
Beim Vorliegen eines solchen Fehlers sollte auch ein Abbruch der Klassenanalyse in Betracht gezogen werden

Reply:
Das framework bietet keine solche Funktion zum abbrechen außer das komplette abrechen der analyse. Aber das komplette abrechen sollte am besten nicht geschehen.